### ISSUE TITLE
[Tune] [Docs] Checkpointing example resume functionality is misleading

### ISSUE THREAD

**Issue by justinvyu** [ray-project/ray#28722]
### Description

This [example](https://docs.ray.io/en/master/tune/tutorials/tune-checkpoints.html#a-simple-cloud-checkpointing-example) doesn't seem to actually resume the experiment as one of the comments suggest:

> `# if this experiment exists, we will resume from the last run`
>  `# as specified by`
>  `name="my-tune-exp",`

Recreating a `Tuner(…)` rather than going through `Tuner.restore(...)` currently never resumes (`resume=False` is being passed into `tune.run` -> `TrialRunner`), and there’s no way to pass in a resume config into `Tuner`.

The [old docs had the same example](https://docs.ray.io/en/releases-1.13.0/tune/tutorials/tune-checkpoints.html#a-simple-cloud-checkpointing-example), and the code here does work because it manually passed in `resume="AUTO"`, but this is not configured in the current broken example with the new 2.0 API.

In summary, the only way to resume a Tune experiment is through `Tuner.restore` right now, and the docs should not imply that there is another way. This makes it so that a change in your run script is needed to resume from an errored experiment (i.e. changing `tuner = Tuner(...)` to `tuner = Tuner.restore(exp_path, resume_errored=True)`).

### Link

https://docs.ray.io/en/master/tune/tutorials/tune-checkpoints.html#a-simple-cloud-checkpointing-example
